rtmbot/core.py

```diff
@@ -183,23 +223,21 @@ def do_jobs(self):
     def do_output(self):
         output = []
         while True:
-            if 'outputs' in dir(self.module):
-                if len(self.module.outputs) > 0:
-                    logging.info("output from {}".format(self.module))
-                    output.append(self.module.outputs.pop(0))
-                else:
-                    break
+            if len(self.outputs) > 0:
+                logging.info("output from {}".format(self.name))
+                output.append(self.outputs.pop(0))
             else:
-                self.module.outputs = []
+                break
         return output
 
 
 class Job(object):
-    def __init__(self, interval, function, debug):
+    def __init__(self, interval, function, debug, plugin):
```

Comment:
From the documentation, it looks like the primary reason to store a reference to the plugin is to obtain a reference to the `SlackClient` in order to interact with the RTM. If so, how about about passing the `SlackClient` instance to the scheduled `Job` functions instead? This would cut down on the state that `Job` need to track.